model/DataObject.php

```diff
@@ -3403,13 +3404,58 @@ public function databaseIndexes() {
 			}
 		}
 
+		if ($sort && is_string($sort)) {
+			$sort = preg_split('/,(?![^()]*+\\))/', $sort);
+
+			foreach ($sort as $value) {
+				try {
+					list ($table, $column) = $this->parseSortColumn(trim($value));
+
+					if ($table && strtolower($table) !== strtolower($this->class)) {
+						continue;
+					}
+
+					if (substr($column, 0, 1) === '"') {
+						$column = substr($column, 1, strlen($column)-1);
+					}
+
+					if (substr($column, -1, 1) === '"') {
+						$column = substr($column, 0, -1);
+					}
```

Comment:
Would `trim($column, '"')` have the same effect as these two `substr` blocks?

Reply:
hah, that would be a touch more elegant, wouldn't it!